### ISSUE TITLE
Newb cannot run downloader

### ISSUE THREAD

**Issue by atax112** [jbazant/zonerama-downloader#1]
Hi.
Running win11
I have installed node.js and git for windows.

From cmd installed yarn, `npm install --global yarn
`
I have cloned the repo with desktop app, have it in C:\Users\Skullbot\Documents\GitHub\zonerama-downloader

Getting error:

`yarn run v1.22.21
$ ts-node index.ts *REDACTED LINK*
'ts-node' is not recognized as an internal or external command,
operable program or batch file.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.`

What am I doing wrong?

**Comment by atax112** [comment]
Installed ts-node with `npm install -g ts-node`

Now this:

```

TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for C:\Users\Skullbot\Documents\GitHub\zonerama-downloader\index.ts
    at new NodeError (node:internal/errors:406:5)
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:99:9)
    at defaultGetFormat (node:internal/modules/esm/get_format:142:36)
    at defaultLoad (node:internal/modules/esm/load:120:20)
    at nextLoad (node:internal/modules/esm/hooks:833:28)
    at load (C:\Users\Skullbot\AppData\Roaming\npm\node_modules\ts-node\dist\child\child-loader.js:19:122)
    at nextLoad (node:internal/modules/esm/hooks:833:28)
    at Hooks.load (node:internal/modules/esm/hooks:416:26)
    at MessagePort.handleMessage (node:internal/modules/esm/worker:168:24)
    at [nodejs.internal.kHybridDispatch] (node:internal/event_target:807:20) {
  code: 'ERR_UNKNOWN_FILE_EXTENSION'
```

**Comment by jbazant** [comment]
Hi, I am a bit surprised someone is actually trying to use this utility. Unfortunately I am unable to try it on windows.

But basic usage is described in Readme:
1. clone repo
2. Call `yarn install` inside the project directory (should be `C:\Users\Skullbot\Documents\GitHub\zonerama-downloader` in your case)
3. After that you should be able to call `yarn start` to execute script (again from project directory)

**Comment by atax112** [comment]
:D Nazdar, som z Kosic inac.

I will try again... not sure, but its the same over and over, unknown extension

**Comment by atax112** [comment]
So googled a bit and found `node --loader ts-node/esm`, but now this error

```
C:\Users\Skullbot\Documents\GitHub\zonerama-downloader>node --loader ts-node/esm index.ts
(node:27808) ExperimentalWarning: `--experimental-loader` may be removed in the future; instead use `register()`:
--import 'data:text/javascript,import { register } from "node:module"; import { pathToFileURL } from "node:url"; register("ts-node/esm", pathToFileURL("./"));'
(Use `node --trace-warnings ...` to show where the warning was created)
? Enter the url of the album *REDACTED*
Fetching gallery cookie...
DONE
Downloading gallery data...
DONE
Error: Cannot read properties of undefined (reading 'map')
```

**Comment by jbazant** [comment]
> So googled a bit and found `node --loader ts-node/esm`

You should be able to run it without those options. but if it works for you, then use it 🤷 . According to your log, it definitely helped.

I just use something like this 
```
~/Projects/personal/zonerama-downloader> yarn start  https://eu.zonerama.com/ipphotool/Album/10709843
```

> but now this error

It looks like no gallery images were returned. Are you sure you specified correct URL? In case the gallery is protected you need to specify `secret` as well (URL will be like `https://eu.zonerama.com/username/Album/123456?secret=ABC123`)

I updated error logging a bit, if you use current version maybe it helps you to identify your problem. (But the script is still pretty dumb honestly)

**Comment by atax112** [comment]
Alright, 

I pulled the repo and now: 

`Error: Invalid gallery json response: "\r\n<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Strict//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\">\r\n<html class=\"unknown\" xmlns='http://www.w3.org/1999/xhtml' xmlns:fb=\"http:/...`

**Comment by jbazant** [comment]
As I said before. This looks like you did not provide the correct secret in the URL.

**Comment by atax112** [comment]
Yeah, I understand, its just the URL in the address bar when looking at the gallery soooo I dont know...

Anyway, thanks for your effort and the script, its a great idea :)

**Comment by jbazant** [comment]
Interesting. I just tried it right now and it worked for me for both public and protected albums ,🤷‍♂️

**Comment by atax112** [comment]
Ok, no way around it.

Try
https://eu.zonerama.com/kittuuussphoto/Album/10708926?secret=C0C34Fn99k8OWKHW4516CyeeQ

**Comment by jbazant** [comment]
Ah I see. It is also password protected one. I will try to add this functionality to the script and reach to you, hopefully by tomorrow evening.

**Comment by atax112** [comment]
Ohhh, now I get it :D

ok, many thanks

**Comment by jbazant** [comment]
I was able to get to it today. Please pull last version of the script and try again.

**Comment by atax112** [comment]
Asks for password but same error:

`Unlocking album...
DONE
Downloading gallery data...
Error: Invalid gallery json response: "\r\n<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Strict//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\">\r\n<html class=\"unknown\" xmlns='http://www.w3.org/1999/xhtml' xmlns:fb=\"http:/...`

**Comment by jbazant** [comment]
Here is my testing album: https://eu.zonerama.com/jbazant/Album/10712711. Pasword is `heslotest`. Does script work for this one?

Have you tried to specify album without secret?

**Comment by atax112** [comment]
I have tried yours, it works.

I have also tried mine without secret: `https://eu.zonerama.com/kittuuussphoto/Album/10708926`
with secret: `https://eu.zonerama.com/kittuuussphoto/Album/10708926?secret=C0C34Fn99k8OWKHW4516CyeeQ`

Same invalid gallery json response...

**Comment by jbazant** [comment]
I am out of ideas. Sorry.

**Comment by atax112** [comment]
I assume it might be some dependency, like our environments arent the same above the obvious win/unix difference...

Not sure how to make a clean install and why doesnt yarn/ts-node work unless I use `node --loader ts-node/esm index.ts`

Anyway, many thanks for the efforts and a great approach. If I figure it out, Ill let you know.

**Comment by atax112** [comment]
Ill try WSL Ubuntu and see if it works there...

**Comment by jbazant** [comment]
This is, in my opinion, caused by some settings of that zonenerama gallery. Changing system will not help.

But honestly, if your only concern is that 37 photos, you can just download them manually.

**Comment by atax112** [comment]
I agree, but I like to figure things out. 

This is a good hlavolam and I might learn something 😁
